src/store/modules/wallet.js

```diff
@@ -672,6 +680,46 @@ const actions = {
     return memo;
   },
 
+  async fetchSuggestedFollowList({ state, commit }) {
+    const currentFollowees = state.followees;
+    const currentFolloweesSet = new Set(currentFollowees);
+
+    // Cache check and return cached result if it's valid
+    if (
+      state.suggestedFollowList.length &&
+      !currentFollowees.some(user => state.suggestedFollowList.includes(user))
+    ) {
+      return state.suggestedFollowList;
+    }
+
+    const initialSuggestions = [...DEFAULT_SUGGESTED_FOLLOW_LIST].filter(
+      suggestion => !currentFolloweesSet.has(suggestion)
+    );
+
+    const suggestedSet = new Set(initialSuggestions);
+    let currentIndex = 0;
+
+    // If we don't have enough suggestions, fetch more
+    while (suggestedSet.size < 3 && currentIndex < currentFollowees.length) {
+      // eslint-disable-next-line no-await-in-loop
+      const { followees = [], pastFollowees = [] } = await this.$axios.$get(
+        getUserV2Followees(currentFollowees[currentIndex])
+      );
+
+      const followSuggestions = [...followees, ...pastFollowees].filter(
+        f => !currentFolloweesSet.has(f)
+      );
+
+      followSuggestions.forEach(suggestion => suggestedSet.add(suggestion));
+
+      currentIndex += 1;
+    }
```

Comment:
Should handle in server side since other user's followee information should not be public